### PR TITLE
RELENG-5886 check that the token is related to the username given

### PR DESCRIPTION
**Pull request by bertranddemiddelaer**

### CODE REVIEW

**Comment by bertranddemiddelaer** [comment]
Will have to fix the fake github server for tests too

File: lua/github_access.lua [hunk 1]
```diff
@@ -105,21 +105,27 @@ end
 
 function check_github(auth_md5, username)
     ngx.log(ngx.STDERR, "checking github for " .. username .. " (" .. auth_md5  .. ")")
-    local res =  ngx.location.capture("/force_github_request/orgs/" .. github_api_company .. "/members/" .. username)
+
+    -- authenticate
+    --
+    local res = ngx.location.capture("/force_github_request/user")
```
Comment (tcarmet) [inline lua/github_access.lua:111]: I double checked if we ask the user to create a token with `read:user` we do. All good, no announcement needed :D

**Review by tcarmet — APPROVED** [review]

**Comment by tcarmet** [comment]
> Will have to fix the fake github server for tests too in order to support another API endpoint

aah yes, /user?

**Comment by bertranddemiddelaer** [comment]
> > Will have to fix the fake github server for tests too in order to support another API endpoint
> 
> aah yes, /user?

Yep :)

**Pull request merged**